src/server/scripts/Northrend/Naxxramas/boss_faerlina.cpp

```diff
@@ -192,19 +208,40 @@ class npc_faerlina_add : public CreatureScript
 
             void Reset() override
             {
-                if (GetDifficulty() == RAID_DIFFICULTY_10MAN_NORMAL) {
+                if (!Is25ManRaid()) {
                     me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_BIND, true);
                     me->ApplySpellImmune(0, IMMUNITY_MECHANIC, MECHANIC_CHARM, true);
                 }
             }
 
+            void EnterCombat(Unit* /*who*/) override
+            {
+                if (_instance)
+                    if (Creature* faerlina = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_FAERLINA)))
+                        faerlina->AI()->DoZoneInCombat(nullptr, 250.0f);
+            }
+
             void JustDied(Unit* /*killer*/) override
             {
-                if (_instance && GetDifficulty() == RAID_DIFFICULTY_10MAN_NORMAL)
+                if (!Is25ManRaid())
                     if (Creature* faerlina = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_FAERLINA)))
                         DoCast(faerlina, SPELL_WIDOWS_EMBRACE);
             }
 
+            void UpdateAI(uint32 /*diff*/) override
+            {
+                if (!UpdateVictim())
+                    return;
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+
+                DoCastVictim(SPELL_ADD_FIREBALL);
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+
+                DoMeleeAttackIfReady();
```

Comment:
It's accessible if the Fireball cast fails for some reason (out of range, no mana, school locked out, etc etc).